apps/rush-lib/src/logic/deploy/DeployArchiver.ts

```
@@ -0,0 +1,97 @@
+import * as JSZip from 'jszip';
+import * as path from 'path';
+import { FileSystem, FileSystemStats, AlreadyExistsBehavior } from '@rushstack/node-core-library';
+import { IDeployState } from './DeployManager';
+
+// JSZip is dependant on Blob being declared.
+declare global {
+  // eslint-disable-next-line
+  type Blob = any;
+}
+export class DeployArchiver {
+  public static async createArchive(deployState: IDeployState): Promise<void> {
+    if (deployState.scenarioConfiguration.json.postCopySourceFolder !== undefined) {
+      const sourceFolderPath: string = FileSystem.getRealPath(
+        deployState.scenarioConfiguration.json.postCopySourceFolder
+      );
+      FileSystem.copyFiles({
+        sourcePath: sourceFolderPath,
+        destinationPath: deployState.targetRootFolder,
+        alreadyExistsBehavior: AlreadyExistsBehavior.Error
+      });
+    }
+    if (deployState.createArchiveFileName !== undefined) {
+      if (!deployState.createArchiveFileName.endsWith('.zip')) {
+        console.log(
```

This should probably throw an exception